store/rootmulti/store.go

```diff
@@ -541,9 +543,29 @@ func (rs *Store) CacheMultiStoreWithVersion(version int64) (types.CacheMultiStor
 			// version does not exist or is pruned, an error should be returned.
 			var err error
 			cacheStore, err = store.(*iavl.Store).GetImmutable(version)
+			// if we got error from loading a module store
+			// we fetch commit info of this version
+			// we use commit info to check if the store existed at this version or not
 			if err != nil {
-				return nil, err
+				if commitInfo == nil {
+					commitInfo, errCommitInfo := rs.GetCommitInfo(version)
```

Comment:
commitInfo will always be nil, as @alexanderbez said, because this assignment uses `:=` and therefore shadows the outer variable. [See this example](https://go.dev/play/p/V7PoCPor_zv).

Reply:
Ahh yes, excellent catch. I also missed the `:=`. Please update @catShaark and we can merge 🙏